### ISSUE TITLE
[PPC] Unit test failing in Geometry/DTGeometryBuilder

### ISSUE THREAD

**Issue by mrodozov** [cms-sw/cmssw#32089]
This unit test:
GeometryDTGeometryBuilderTestDriver
is failing:
`Failure using cmsRun /scratch/cmsbuild/jenkins_a/workspace/ib-run-qa/CMSSW_11_2_X_2020-11-05-2300/src/Geometry/DTGeometryBuilder/test/python/validateDTGeometry_cfg.py: status 139`
in powerpc IBs
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_ppc64le_gcc820/CMSSW_11_2_X_2020-11-05-2300/unitTestLogs/Geometry/DTGeometryBuilder#/83

**Comment by makortel** [comment]
assign geometry

**Comment by makortel** [comment]
Some more details from the log in the description
```
#3  0x0000100009924838 in sig_dostack_then_abort () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginFWCoreServicesPlugins.so
#4  <signal handler called>
#5  0x00001000029e73d8 in __kernel_sincosl () from /lib64/libm.so.6
#6  0x00001000029dffa8 in sincosl@@GLIBC_2.17 () from /lib64/libm.so.6
#7  0x0000100225a44f8c in algorithm(dd4hep::Detector&, cms::DDParsingContext&, dd4hep::xml::Handle_t) [clone .isra.136] () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDD4hep_GeometryEcalCommonDataPlugins.so
#8  0x000010022597fad8 in void dd4hep::xml::Collection_t::for_each<dd4hep::Converter<dd4hep::(anonymous namespace)::DDLAlgorithm, dd4hep::xml::Handle_t> >(dd4hep::Converter<dd4hep::(anonymous namespace)::DDLAlgorithm, dd4hep::xml::Handle_t>) const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDD4HepPlugins.so
#9  0x000010022598d6b8 in load_dddefinition(dd4hep::Detector&, dd4hep::xml::Handle_t) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDD4HepPlugins.so
#10 0x000010022598ec34 in (anonymous namespace)::Factory<dd4hep::(anonymous namespace)::xml_document_DDDefinition, long (dd4hep::Detector*, dd4hep::xml::Handle_t*)>::call(dd4hep::Detector*, dd4hep::xml::Handle_t*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDD4HepPlugins.so
#11 0x000010001f546668 in dd4hep::DetectorLoad::processXMLElement(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::xml::Handle_t const&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#12 0x000010001f5473ac in dd4hep::DetectorLoad::processXML(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::xml::UriReader*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#13 0x000010001f53b3bc in non-virtual thunk to dd4hep::DetectorImp::fromCompact(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::DetectorBuildType) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#14 0x00001002258f9444 in (anonymous namespace)::Factory<dd4hep::(anonymous namespace)::DD4hep_CompactLoader, long (dd4hep::Detector*, int, char**)>::call(dd4hep::Detector*, int, char**) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCorePlugins.so.1.14
#15 0x000010001f53bb48 in long dd4hep::PluginService::Create<long, dd4hep::Detector*, int, char**>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::Detector*, int, char**) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#16 0x000010001f5340e8 in dd4hep::DetectorImp::apply(char const*, int, char**) const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#17 0x000010001ee98144 in cms::DDDetector::process(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDDCMS.so
#18 0x000010001ee98b44 in cms::DDDetector::DDDetector(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDDCMS.so
#19 0x0000100221c56420 in DDDetectorESProducer::produceGeom(IdealGeometryRecord const&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#20 0x0000100221c64d84 in decltype ({parm#1}()) edm::convertException::wrap<edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}::operator()() const::{lambda()#1}>(edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}::operator()() const::{lambda()#1}) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#21 0x0000100221c652a8 in edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}::operator()() const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#22 0x0000100221c66684 in void edm::SerialTaskQueueChain::actionToRun<edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}&>(edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#23 0x0000100221c66788 in edm::SerialTaskQueue::QueuedTask<edm::SerialTaskQueueChain::push<edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}>(edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}&&)::{lambda()#1}>::execute() () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#24 0x00001000024ab780 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::process_bypass_loop (this=0x1000037ff200, context_guard=..., t=0x1000037f1c40, isolation=0) at ../../include/tbb/task.h:1003
#25 0x00001000024abaa4 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all (this=0x1000037ff200, parent=..., child=<optimized out>) at ../../include/tbb/task.h:1003
#26 0x00001000024ac4b0 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::wait_for_all (this=<optimized out>, parent=..., child=0x0) at ../../src/tbb/governor.h:123
#27 0x00001000003a134c in tbb::interface7::internal::delegated_function<edm::Worker::esPrefetchAsync(edm::WaitingTask*, edm::EventSetupImpl const&, edm::Transition, edm::ServiceToken const&)::{lambda()#1}::operator()()::{lambda()#1} const, void>::operator()() const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#28 0x00001000024a4be8 in tbb::interface7::internal::task_arena_base::internal_execute (this=0x1000004e6ce0 <edm::esTaskArena()::s_arena>, d=...) at ../../src/tbb/arena.cpp:1105
#29 0x00001000003a0358 in edm::FunctorTask<edm::Worker::esPrefetchAsync(edm::WaitingTask*, edm::EventSetupImpl const&, edm::Transition, edm::ServiceToken const&)::{lambda()#1}>::execute() () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
```

**Comment by mrodozov** [comment]
fixed by #33705
